iOS_SDK/OneSignalDevApp/OneSignalDevApp/ViewController.m

```diff
@@ -96,14 +96,10 @@ - (IBAction)sendTagButton:(id)sender {
                  NSLog(@"failures: %d", ++failures);
     }];
     
-    [OneSignal IdsAvailable:^(NSString *userId, NSString *pushToken) {
-        NSLog(@"IdsAvailable Fired");
-    }];
```

Comment:
How do we know now when the `Ids` are `Available`? What replaces this callback and how do we ensure someone is not trying to perform tasks with an unregistered device?

Reply:
The customers implement  `getPermissionSubscriptionState`, `addPermissionObserver`, & `addSubscriptionObserver` instead